kernel/packages/shared/atlas/reducer.ts

```diff
@@ -30,68 +42,130 @@ export function atlasReducer(state?: AtlasState, action?: AnyAction) {
     case MARKET_DATA:
-      const marketAction = action as PayloadAction<typeof MARKET_DATA, MarketData>
-      return {
-        ...state,
-        marketName: {
-          ...state.marketName,
-          ...marketAction.payload.data
-        }
-      }
+      return reduceMarketData(state, action.payload)
     case REPORTED_SCENES_FOR_MINIMAP:
-      return {
-        ...state,
-        lastReportPosition: action.payload.reportPosition ? action.payload.reportPosition : state.lastReportPosition,
-        atlasReducer: {
-          ...state.alreadyReported,
-          ...action.payload.parcels.reduce((prev: Record<string, boolean>, next: string) => {
-            prev[next] = true
-            return prev
-          }, {})
-        }
-      }
+      return reduceReportedScenesForMinimap(state, action.payload)
     case DISTRICT_DATA:
-      return {
-        ...state,
-        districtName: {
-          ...state.districtName,
-          ...zip(action.payload.data, (t: District) => [t.id, t.name])
-        }
-      }
+      return reduceDistrictData(state, action)
+  }
+  return state
+}
+
+function reduceFetchDataFromSceneJson(state: AtlasState, sceneId: string) {
+  if (!state.idToScene[sceneId]) {
+    state.idToScene[sceneId] = { ...MAP_SCENE_DATA_INITIAL_STATE }
+  }
+
+  state.idToScene[sceneId].requestStatus = 'loading'
+  return state
+}
+
+function reduceFailureDataFromSceneJson(state: AtlasState, sceneId: string) {
+  if (!state.idToScene[sceneId]) {
+    state.idToScene[sceneId] = { ...MAP_SCENE_DATA_INITIAL_STATE }
   }
+
+  state.idToScene[sceneId].requestStatus = 'fail'
   return state
 }
+
+function reduceSuccessDataFromSceneJson(state: AtlasState, landData: ILand) {
+  const tileToScene = { ...state.tileToScene }
+  const idToScene = { ...state.idToScene }
+
+  let mapScene: MapSceneData = { ...state.idToScene[landData.sceneId] }
+
+  // NOTE(Brian): this code is for the case in which market data comes first (most likely always)
+  //             in that case we find the tileToScene data and update the mapScene with the
+  //             relevant market values. If we don't do this we will get an inconsistent state.
+  landData.sceneJsonData.scene.parcels.forEach(x => {
+    if (state.tileToScene[x]) {
+      mapScene = {
+        ...mapScene,
+        name: state.tileToScene[x].name,
+        type: state.tileToScene[x].type,
+        estateId: state.tileToScene[x].estateId
+      }
+    }
+  })
+
+  mapScene.requestStatus = 'ok'
+  mapScene.sceneJsonData = landData.sceneJsonData
+
+  const name = getSceneNameFromAtlasState(state, mapScene.sceneJsonData) ?? mapScene.name
+  mapScene.name = postProcessSceneName(name)
+
+  mapScene.sceneJsonData.scene.parcels.forEach(x => {
+    tileToScene[x] = mapScene
+  })
+
+  idToScene[landData.sceneId] = mapScene
+
+  return { ...state, tileToScene, idToScene }
+}
+
+function reduceDistrictData(state: AtlasState, action: AnyAction) {
+  state.hasDistrictData = true
+  return state
```

Comment:
Remember to remove those comments. On a sidenote, district fetching should remain as a stub for now because we aren't used from unity side yet.